packages/venia-ui/lib/components/CheckoutPage/PaymentInformation/summary.js

```diff
@@ -43,73 +39,23 @@ const Summary = props => {
         );
     }
 
-    if (selectedPaymentMethod.code !== 'braintree') {
+    const hasCustomSummaryComp = Object.keys(summaryPayments).includes(
+        selectedPaymentMethod.code
+    );
+
+    if (hasCustomSummaryComp) {
+        const SummaryPaymentMethodComponent =
+            summaryPayments[selectedPaymentMethod.code];
         return (
-            <div className={classes.root}>
-                <div className={classes.heading_container}>
-                    <h5 className={classes.heading}>
-                        <FormattedMessage
-                            id={'checkoutPage.paymentInformation'}
-                            defaultMessage={'Payment Information'}
-                        />
-                    </h5>
-                </div>
-                <div className={classes.card_details_container}>
-                    <span className={classes.payment_details}>
-                        {/* TODO: Do we want to display some other info? On mobile this can feel odd since price summary isn't seen until you go to the review step. */}
-                        {selectedPaymentMethod.title}
-                    </span>
-                </div>
-            </div>
+            <SummaryPaymentMethodComponent
+                selectedPaymentMethod={selectedPaymentMethod}
+                billingAddress={billingAddress}
+                paymentNonce={paymentNonce}
+                isBillingAddressSame={isBillingAddressSame}
+                onEdit={onEdit}
```

Comment:
`paymentNonce` is specific to the braintree implementation summary. I wonder if instead we should pass something like a `details` object that can have any props (specific to the method). You could obtain these details by modifying the `useSummary` talon to return them, or you could just have the talon return a general bucket of details that the summary can pick from though that might surface a risk vector.

```suggestion
                selectedPaymentMethod={selectedPaymentMethod}
                billingAddress={billingAddress}
                paymentDetails={detailsFromSummaryTalon}
                isBillingAddressSame={isBillingAddressSame}
                onEdit={onEdit}
```

Reply:
Resolved in 4af14bc07567ced352aafdffee4140b0ec0c6b0f